### PR TITLE
fix home/header login button

### LINKED ISSUE #227: fix home/header login button
One button on the login page and a logout button on the rest of the pages in the header

### PR DESCRIPTION
**Pull request by warreprovoost**
Closes #227 #241

### CODE REVIEW

File: frontend/src/utils/fetches/FetchMe.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+export const fetchMe = async () => {
+  const API_URL = import.meta.env.VITE_APP_API_HOST
+  try {
+    const response = await fetch(`${API_URL}/me`, {
+      credentials: 'include'
+    })
+    if(response.status == 200){
```
Comment (Vucis) [inline frontend/src/utils/fetches/FetchMe.ts:8]: stay consistent with spaces: if() and then later catch () and then }else {

**Review by Vucis — APPROVED** [review]

File: frontend/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -44,32 +49,70 @@ export function Header(): JSX.Element {
...
+
+  React.useEffect(() => {
     const baseItems = [{ link: "/", text: t("homepage") }];
     const additionalItems = [
       { link: "/projects", text: t("myProjects") },
-      { link: "/courses", text: t("myCourses") }
+      { link: "/courses", text: t("myCourses") },
     ];
-    if (isLoggedIn()) {
+    if (me.role !== "UNKNOWN") {
```
Comment (AronBuzogany) [inline frontend/src/components/Header/Header.tsx:69]: No, this must check if person is logged in, don't check the role. Or check the role inside the `isLoggedIn` function

File: frontend/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -44,32 +49,70 @@ export function Header(): JSX.Element {
...
+              <Menu
+                anchorEl={anchorEl}
+                open={Boolean(anchorEl)}
+                onClose={() => setAnchorEl(null)}
+              >
+                <Typography sx={{ padding: "6px 16px", color: "black" }}>
+                  {me.display_name}
+                </Typography>
+                <MenuItem
+                  onClick={() => (window.location.href = `${API_URL}/logout`)}
```
Comment (AronBuzogany) [inline frontend/src/components/Header/Header.tsx:109]: Don't do `window.location.href` there are functions made for redirects, `navigate`

File: frontend/src/components/Header/Login.tsx [hunk 4]
```diff
@@ -11,6 +12,12 @@ const TENANT_ID = import.meta.env.VITE_APP_TENANT_ID;
  */
 export function LoginButton(): JSX.Element {
   const link = `https://login.microsoftonline.com/${TENANT_ID}/oauth2/v2.0/authorize?prompt=select_account&response_type=code&client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&scope=.default`;
+  const { t } = useTranslation("translation", { keyPrefix: "home" });
 
-  return <Button variant="contained" component={Link} to={link}>Login</Button>
+  return (
+    <Button variant="contained" component={Link} to={link}>
+      {" "}
```
Comment (AronBuzogany) [inline frontend/src/components/Header/Login.tsx:19]: What's this for

File: frontend/src/pages/home/HomePages.tsx [hunk 5]
```diff
@@ -2,18 +2,19 @@ import HomePage from './HomePage.tsx';
...
-    projects: ProjectDeadline[],
-    me: string
-  }
-  const me = loader.me
-  if (me === 'UNKNOWN') {
+    projects: ProjectDeadline[];
+    me: Me;
+  };
+  const me = loader.me.role;
+  if (me === "UNKNOWN") {
```
Comment (AronBuzogany) [inline frontend/src/pages/home/HomePages.tsx:17]: Same here, make a function that checks if someone is logged in, the meaning of roles can change which can cause this to break

File: frontend/src/utils/fetches/FetchMe.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  const API_URL = import.meta.env.VITE_APP_API_HOST;
+  try {
+    const response = await fetch(`${API_URL}/me`, {
+      credentials: "include",
+    });
+    if (response.status == 200) {
+      const data = await response.json();
+      return data.data;
+    } else {
+      return { role: "UNKNOWN" };
```
Comment (AronBuzogany) [inline frontend/src/utils/fetches/FetchMe.ts:11]: you can change this to `loggedIn: false` for example, and above `true`

File: frontend/src/components/Header/Header.tsx [hunk 7]
```diff
@@ -44,32 +49,70 @@ export function Header(): JSX.Element {
...
+            edge="start"
+            onClick={() => setOpen(!open)}
+            sx={{ color: "white", marginLeft: 0 }}
+          >
+            <MenuIcon style={{ fontSize: "2rem" }} />
           </IconButton>
-          <TitlePortal/>
-          <LoginButton></LoginButton>
+          <TitlePortal />
+          {me.role !== "UNKNOWN" && (
```
Comment (AronBuzogany) [inline frontend/src/components/Header/Header.tsx:88]: Keep the login button in the header also, that way people can log in from `404` for example page also

File: frontend/src/components/Header/Header.tsx (hunk 2 above)
Comment (AronBuzogany) [reply frontend/src/components/Header/Header.tsx]: Not solved

File: frontend/src/components/Header/Header.tsx (hunk 3 above)
Comment (AronBuzogany) [reply frontend/src/components/Header/Header.tsx:113]: not fixed

**Review by AronBuzogany — CHANGES_REQUESTED** [review]

File: frontend/src/components/Header/Header.tsx [hunk 8]
```diff
@@ -109,10 +109,9 @@ export function Header({ me }: HeaderProps): JSX.Element {
                 <Typography sx={{ padding: "6px 16px", color: "black" }}>
                   {me.display_name}
                 </Typography>
-                <MenuItem
-                  onClick={() => (window.location.href = `${API_URL}/logout`)}
-                >
-                  {t("logout")}
+                <MenuItem>
```
Comment (AronBuzogany) [inline frontend/src/components/Header/Header.tsx:112]: Next time you can do

`<MenuItem component="a" href="..." >`

**Review by AronBuzogany — APPROVED** [review]

**Pull request merged**